support/cas-server-support-aup-ldap/src/main/java/org/apereo/cas/aup/LdapAcceptableUsagePolicyRepository.java

```diff
@@ -30,12 +32,17 @@
  * @since 4.2
  */
 @Slf4j
-public class LdapAcceptableUsagePolicyRepository extends BaseAcceptableUsagePolicyRepository {
+public class LdapAcceptableUsagePolicyRepository extends BaseAcceptableUsagePolicyRepository implements DisposableBean {
     private static final long serialVersionUID = 1600024683199961892L;
 
+    private HashMap<LdapAcceptableUsagePolicyProperties, ConnectionFactory> connectionFactoryList = new HashMap<>();
+
     public LdapAcceptableUsagePolicyRepository(final TicketRegistrySupport ticketRegistrySupport,
                                                final AcceptableUsagePolicyProperties aupProperties) {
         super(ticketRegistrySupport, aupProperties);
+        aupProperties.getLdap().forEach(ldap -> {
+            connectionFactoryList.put(ldap, LdapUtils.newLdaptiveConnectionFactory(ldap));
+        });
```

Comment:
Pass the connection factory map as argument; build inside the bean in config class.

Reply:
See above.